### PR TITLE
nzbget: switch to nzbgetcom version, update build flow and dependencies

### PR DESCRIPTION
**Pull request by phnzb**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Original NZBGet was abandoned years ago. NZBGet by nzbget.com is a spiritual successor of the original project, widely acknowledged by the community. We are actively developing it and have already released 3 versions - with fixes and with new features like ExtensionManager.

### CODE REVIEW

File: Formula/n/nzbget.rb [hunk 1]
```diff
@@ -20,58 +20,39 @@ class Nzbget < Formula
...
-      (buildpath/"brew_include").install_symlink MacOS.sdk_path/"usr/include/ncurses.h"
-      ENV["ncurses_CFLAGS"] = "-I#{buildpath}/brew_include"
-      ENV["ncurses_LIBS"] = "-L/usr/lib -lncurses"
-    else
-      ENV["ncurses_CFLAGS"] = "-I#{Formula["ncurses"].opt_include}"
-      ENV["ncurses_LIBS"] = "-L#{Formula["ncurses"].opt_lib} -lncurses"
-    end
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    system "cmake", "--build", "build"
+    system "strip", "build/nzbget"
```
Comment (chenrui333) [inline Formula/n/nzbget.rb:34]: can you move this to your cmake build config?

File: Formula/n/nzbget.rb [hunk 2]
```diff
@@ -20,58 +20,39 @@ class Nzbget < Formula
...
-    # Tell configure to use OpenSSL
-    system "./configure", "--disable-debug", "--disable-dependency-tracking",
-                          "--prefix=#{prefix}",
-                          "--with-tlslib=OpenSSL"
-    system "make"
-    ENV.deparallelize
-    system "make", "install"
-    pkgshare.install_symlink "nzbget.conf" => "webui/nzbget.conf"
+    # remove default nzbget.conf to prevent linking
+    rm prefix/"etc/nzbget.conf"
```
Comment (chenrui333) [inline Formula/n/nzbget.rb:38]: this can be part of cmake file as well

File: Formula/n/nzbget.rb (hunk 1 above)
Comment (phnzb) [reply Formula/n/nzbget.rb:34]: We have to make neccessary changes to nzbget first to comply, we'll be happy to do that with v25 (next release). v24 has just been released and can't be easily updated.

Comment (chenrui333) [reply Formula/n/nzbget.rb:34]: Any issue for tracking this, can you add that as code comments? Thanks!

Comment (phnzb) [reply Formula/n/nzbget.rb:37]: Tracking issue https://github.com/nzbgetcom/nzbget/issues/257 - comments added

Comment (chenrui333) [reply Formula/n/nzbget.rb:37]: thanks a lot!

**Review by chenrui333 — DISMISSED** [review]

**Comment by chenrui333** [comment]
@phnzb, thanks for your first contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**